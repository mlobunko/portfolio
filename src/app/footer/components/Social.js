import React from "react";
import PropTypes from "prop-types";

import { StyledSocial } from "./styles";

export const Social = ({ link, name, iconComponent }) => (
  <a
    href={link}
    target={"_blank"}
    rel={"noopener noreferrer"}
    aria-label={name}
  >
    <StyledSocial>{iconComponent}</StyledSocial>
  </a>
);
Social.displayName = "Social";

export default Social;

Social.propTypes = {
  iconComponent: PropTypes.element.isRequired,
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};
