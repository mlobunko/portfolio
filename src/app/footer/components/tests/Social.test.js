import React from "react";
import { shallow } from "enzyme";

import { Social } from "../Social";

describe("Social", () => {
  it("renders correctly", () => {
    const props = {
      link: "123",
      name: "family",
      iconComponent: <div />
    };
    const wrapper = shallow(<Social {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
