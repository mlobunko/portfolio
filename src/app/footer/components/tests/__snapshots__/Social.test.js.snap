// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social renders correctly 1`] = `
<a
  aria-label="family"
  href="123"
  rel="noopener noreferrer"
  target="_blank"
>
  <StyledSocial>
    <div />
  </StyledSocial>
</a>
`;
